### PR TITLE
Adds Jetpack Cloud environment badges

### PR DESCRIPTION
**Pull request by ChaosExAnima**
#### Changes proposed in this Pull Request

* Adds an environment badge for Jetpack environments.

#### Testing instructions

* Build Jetpack Cloud via `CALYPSO_ENV=jetpack-cloud-development npm start`.
* Let it build, then visit the dashboard page. See environment badge!
* Build Jetpack Cloud staging.
* Start it and visit the dashboard page. See the environment badge.

Screenshots:
Screen Shot 2020-01-30 at 4 48 40 PM

Screen Shot 2020-01-30 at 5 37 27 PM

Fixes 1156570014567299-as-1159371543675235.

### CODE REVIEW

File: client/server/pages/index.js [hunk 1]
```diff
@@ -352,6 +352,16 @@ function getDefaultContext( request, entrypoint = 'entry-main' ) {
 		context.commitChecksum = getCurrentCommitShortChecksum();
 	}
 
+	if ( calypsoEnv === 'jetpack-cloud-stage' ) {
+		context.badge = 'jetpack-cloud-staging';
+		context.feedbackURL = 'https://github.com/Automattic/wp-calypso/issues/';
```
Comment (tyxla) [inline client/server/pages/index.js]: I wonder if https://github.com/Automattic/wp-calypso/issues/new?labels=Jetpack%20Cloud would even make more sense.

File: client/server/pages/index.js [hunk 2]
```diff
@@ -352,6 +352,16 @@ function getDefaultContext( request, entrypoint = 'entry-main' ) {
...
+	} else if ( calypsoEnv === 'jetpack-cloud-development' ) {
```
Comment (tyxla) [inline client/server/pages/index.js]: Nit: seems like we use `else if` here and not for the other cases. Ideally, we would be consistent for all of those cases, or we would use a `switch` or something.

Comment (ChaosExAnima) [reply client/server/pages/index.js]: Fixed in 24537a74f36310f46814241f30496a548db9ba3a. I debating refactoring, but decided it wasn't worth the energy for a micro optimization like that.

File: client/server/pages/index.js (hunk 1 above)
Comment (ChaosExAnima) [reply client/server/pages/index.js]: Thought about it, and I disagree. For one, the other links are all to the plain issues, not the new issue link, so let's keep that. For another, given the crossover between JP Cloud and Calypso, it'll be harder for someone not familiar with the codebase to know what's one or the other. Better let us triage and label appropriately.

**Pull request merged**